### PR TITLE
Updated Greg dll's and improve dependency injection of IAuthProvider

### PR DESCRIPTION
**Pull request by pboyer**
Updates for https://github.com/DynamoDS/Dynamo/pull/3619

@lukechurch PTAL

### CODE REVIEW

File: src/DynamoRevit/DynamoRevit.cs [hunk 1]
```diff
@@ -117,22 +117,7 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
...
-                        revitDynamoModel = InitializeCoreModel(commandData);
-                        dynamoViewModel = InitializeCoreViewModel(revitDynamoModel);
-
-                        // show the window
-                        InitializeCoreView().Show();
-
-                        TryOpenWorkspaceInCommandData(commandData);
-                        SubscribeViewActivating(commandData);
-                    });
-#endif
```
Comment (pboyer) [inline src/DynamoRevit/DynamoRevit.cs]: Yuck.

File: src/DynamoRevit/DynamoRevit.cs [hunk 2]
```diff
@@ -173,28 +158,15 @@ private static RevitDynamoModel InitializeCoreModel(ExternalCommandData commandD
...
-            return RevitDynamoModel.Start(
-                new RevitDynamoModel.StartConfiguration()
-                {
-                    Preferences = prefs,
-                    DynamoCorePath = corePath,
-                    Context = GetRevitContext(commandData)
-                });
-
-#else
-
```
Comment (pboyer) [inline src/DynamoRevit/DynamoRevit.cs]: But wait, there's more!

File: src/DynamoRevit/DynamoRevit.cs [hunk 3]
```diff
@@ -173,28 +158,15 @@ private static RevitDynamoModel InitializeCoreModel(ExternalCommandData commandD
...
             return RevitDynamoModel.Start(
                 new RevitDynamoModel.StartConfiguration()
                 {
                     Preferences = prefs,
                     DynamoCorePath = corePath,
                     Context = GetRevitContext(commandData),
-                    SchedulerThread = new RevitSchedulerThread(commandData.Application)
+                    SchedulerThread = new RevitSchedulerThread(commandData.Application),
+                    AuthProvider = new RevitOxygenProvider(new DispatcherSynchronizationContext(Dispatcher.CurrentDispatcher))
                 });
```
Comment (pboyer) [inline src/DynamoRevit/DynamoRevit.cs]: This is how we inject the `IAuthProvider` dependency.

File: src/DynamoRevit/DynamoRevit.cs [hunk 4]
```diff
@@ -210,9 +182,6 @@ private static DynamoViewModel InitializeCoreViewModel(RevitDynamoModel revitDyn
                         new RevitWatchHandler(vizManager, revitDynamoModel.PreferenceSettings)
                 });
 
-            revitDynamoModel.PackageManagerClient.RequestAuthentication +=
-                 SingleSignOnManager.RegisterSingleSignOn;
-
```
Comment (pboyer) [inline src/DynamoRevit/DynamoRevit.cs]: This event is no longer needed.

File: src/DynamoRevit/DynamoRevit.cs [hunk 5]
```diff
@@ -251,10 +218,6 @@ private static void InitializeCore(ExternalCommandData commandData)
         {
             if (initializedCore) return;
 
-#if !ENABLE_DYNAMO_SCHEDULER
-            IdlePromise.RegisterIdle(commandData.Application);
-#endif
-
             InitializeAssemblies();
```
Comment (pboyer) [inline src/DynamoRevit/DynamoRevit.cs]: Would be great if @Benglin could clean these up.

File: src/DynamoRevit/Models/SingleSignOnManager.cs [hunk 6]
```diff
@@ -1,49 +0,0 @@
...
-using System.IO;
-using System.Reflection;
-using System.Windows.Threading;
-
-using Greg;
-
-namespace Dynamo.Applications
-{
-    internal static class SingleSignOnManager
-    {
```
Comment (pboyer) [inline src/DynamoRevit/Models/SingleSignOnManager.cs]: Moved to GregRevitAuth.dll

File: src/DynamoRevit/DynamoRevit.cs (hunk 5 above)
Comment (Benglin) [inline src/DynamoRevit/DynamoRevit.cs]: [As you wished](https://github.com/DynamoDS/DynamoRevit/pull/132), @pboyer. Do have a look :)

**Pull request merged**